src/Components/Web.JS/src/Rendering/DomMerging/DomSync.ts

```diff
@@ -43,26 +52,34 @@ export function synchronizeDomContent(destination: CommentBoundedRange | Node, n
     for (let editIndex = 0; editIndex < editsLength; editIndex++) {
       const operation = edits[editIndex];
       switch (operation) {
-        case Operation.Keep:
-          treatAsMatch(nextDestinationNode!, nextNewContentNode!);
+        case Operation.Keep: {
+          const destination = nextDestinationNode!;
+          const source = nextNewContentNode!;
           nextDestinationNode = nextDestinationNode!.nextSibling;
           nextNewContentNode = nextNewContentNode!.nextSibling;
+          treatAsMatch(destination, source);
```

Comment:
These sorts of changes could be eliminated if we had an abstraction over the DOM APIs here that presented a view of the world that incorporates "startmarker ... contents ... endmarker" as a single logical node. I *think* that's something you mentioned you'd be considering doing when we chatted.

Reply:
The existing `LogicalElements` APIs *might* be a sufficient representation, as they already have the concept that marker comments can be used to represent a single logical node with children. We just need to take the static content, find all the marker nodes in them, and upgrade them to be LogicalElements in the same way that `toLogicalRootCommentElement` already does. Then the logic in `DomSync` would need to use the `LogicalElements` APIs when walking the DOM so it sees it with respect to that view of the world.

Part of the reason why `synchronizeDomContent` accepts a `CommentBoundedRange` as a way to define the parent node is for precisely this case - we may want to treat a logical element as being the parent.